src/components/router/Router.brs

```diff
@@ -51,22 +48,45 @@ sub resetHistory(rootPath = "" as String)
   m._history = []
 
   if (rootPath <> "")
-    m._history.push({ path: rootPath, params: {} })
+    rootRoute = CreateObject("roSGNode", "ActivatedRoute")
+    rootRoute.setFields({ path: rootPath })
+
+    m._history.push(rootRoute.getFields())
   end if
 end sub
 
-sub _updateHistory()
-  if (m.top.url = "" OR m.top.activatedRoute.shouldSkip)
-    return
+function _getPreviousRoute(data as Object) as Object
+  isBackJourney = getProperty(data, "isBackJourney", false)
+
+  if (isBackJourney OR m.top.activatedRoute.shouldSkip)
```

Comment:
I wonder if we really need to apply this logic when `m.top.activatedRoute.shouldSkip` is true. This is a different flag than `skipInHistory` and it is not set on back journey, but could be set by the middleware. Meaning that it is possible to skip routes only in the non-back journey (skipped routes are not in the history), and in that case `previousRoute` should stay as is. 

E.g. we navigate from route A to route B, middleware updates route B to be skipped and redirects to route C, then we should not set route B as the `previousRoute` and route A should be still there. If we apply the below logic, then we would update the `previousRoute` with the route A, but it seems to be redundant as it is already there.

Reply:
this is basically what I mean we could do:

```suggestion
  if (m.top.activatedRoute.shouldSkip)
    return m.top.previousRoute
  end if
  
  if (isBackJourney)
```